mathjax3-ts/output/svg/Wrapper.ts

```diff
@@ -0,0 +1,424 @@
+        const mathbackground = attributes.getExplicit('mathbackground') as string;
+        const background = attributes.getExplicit('background') as string;
+        if (mathcolor || color) {
+            adaptor.setAttribute(this.element, 'fill', mathcolor || color);
+            adaptor.setAttribute(this.element, 'stroke', mathcolor || color);
+        }
+        if (mathbackground || background) {
+            let {h, d, w} = this.getBBox();
+            let rect = this.svg('rect', {
+                fill: mathbackground || background,
+                x: 0, y: this.fixed(-d),
+                width: this.fixed(w),
+                height: this.fixed(h + d),
+                'data-bgcolor': true
+            });
+            let child = adaptor.firstChild(this.element);
+            if (child) {
+                adaptor.insert(rect, child);
+            } else {
+                adaptor.append(this.element, rect);
+            }
+        }
+    }
+
+    /**
+     * Copy RDFa, aria, and other tags from the MathML to the SVG output nodes.
+     * Don't copy those in the skipAttributes list, or anything that already exists
+     * as a property of the node (e.g., no "onlick", etc.).  If a name in the
+     * skipAttributes object is set to false, then the attribute WILL be copied.
+     * Add the class to any other classes already in use.
+     */
+    protected handleAttributes() {
+        const attributes = this.node.attributes;
+        const defaults = attributes.getAllDefaults();
+        const skip = SVGWrapper.skipAttributes;
+        for (const name of attributes.getExplicitNames()) {
+            if (skip[name] === false || (!(name in defaults) && !skip[name] &&
+                                         !this.adaptor.hasAttribute(this.element, name))) {
+                this.adaptor.setAttribute(this.element, name, attributes.getExplicit(name) as string);
+            }
+        }
+        if (attributes.get('class')) {
+            this.adaptor.addClass(this.element, attributes.get('class') as string);
+        }
+    }
+
+    /*******************************************************************/
+
+    /**
+     * @param {N} svg         The HTML node whose indentation is to be adjusted
+     * @param {string} align  The alignment for the node
+     * @param {number} shift  The indent (positive or negative) for the node
+     */
+    protected setIndent(svg: N, align: string, shift: number) {
+        if (align === 'center' || align === 'left') {
+            // FIXME
+        }
+        if (align === 'center' || align === 'right') {
+            // FIXME
+        }
+    }
+
+    /**
+     * @param {number} x   The x-offset for the element
+     * @param {number} y   The y-offset for the element
+     * @param {N} eleemnt  The element to be placed
+     */
+    public place(x: number, y: number, element: N = null) {
+        if (!element) {
+            element = this.element;
+        }
```

Comment:
This is only necessary if `x || y` holds. So why not have 
```
if (!(x || y)) return;
```
and then have the logic of the method?